### PR TITLE
Add new E2E test for merchant transforming classic cart to cart block

### PR DESCRIPTION
**Pull request by veljkho**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Added merchant test for transforming classic to block-based cart.

Closes # [637](https://github.com/woocommerce/woocommerce-quality/issues/637)

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Checkout this branch locally
2. Execute: `pnpm test:e2e-pw ./tests/e2e-pw/tests/merchant/create-cart-block.spec.js`
3. Make sure it pass

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/tests/e2e-pw/tests/merchant/create-cart-block.spec.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
+const { test, expect, request } = require( '@playwright/test' );
+const { admin } = require( '../../test-data/data' );
+const { closeWelcomeModal } = require( '../../utils/editor' );
+
+const transformedCartBlockTitle = 'Transformed Cart';
```
Comment (adimoldovan) [inline plugins/woocommerce/tests/e2e-pw/tests/merchant/create-cart-block.spec.js]: Can you use an unique page name, to ensure the right page gets tested on test re-runs? 

```suggestion
const transformedCartBlockTitle = `Transformed Cart ${ Date.now() }`;
```

File: plugins/woocommerce/tests/e2e-pw/tests/merchant/create-cart-block.spec.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+		await allPages.forEach( async ( page ) => {
+			if ( page.title.rendered === transformedCartBlockTitle ) {
+				response = await wpApi.delete( `pages/${ page.id }`, {
+					data: {
+						force: true,
+					},
+				} );
+			}
+		} );
+	} );
```
Comment (adimoldovan) [inline plugins/woocommerce/tests/e2e-pw/tests/merchant/create-cart-block.spec.js]: The page deletion is not working, can you please make it work or remove the code if not?

**Review by adimoldovan — CHANGES_REQUESTED** [review]
Thanks for adding this test @veljkho.
It tested well to me, but I can you please take care of the page not being cleaned-up/unique?

File: plugins/woocommerce/tests/e2e-pw/tests/merchant/create-cart-block.spec.js (hunk 1 above)
Comment (veljkho) [reply plugins/woocommerce/tests/e2e-pw/tests/merchant/create-cart-block.spec.js]: Thanks for reviewing @adimoldovan , added as suggested. 👍

File: plugins/woocommerce/tests/e2e-pw/tests/merchant/create-cart-block.spec.js (hunk 2 above)
Comment (veljkho) [reply plugins/woocommerce/tests/e2e-pw/tests/merchant/create-cart-block.spec.js]: Sure, it was removed in a new commit.

**Review by adimoldovan — APPROVED** [review]
Thanks @veljkho, it looks good!

**Pull request merged**